### ISSUE TITLE
RSS lastBuildDate doesn't match RSS specifications

### ISSUE THREAD

**Issue by Ddcdom** [joomla/joomla-cms#13008]
### Steps to reproduce the issue
Create an RSS feed

### Expected result
The timestamp of the channel's <lastBuildDate> node should show the time that the feed content was last updated.

### Actual result
The timestamp of the channel's <lastBuildDate> node is the time of the page load.

### System information (as much as possible)
Joomla 3.6.4 fresh install

### Additional comments
According to the RSS specification https://validator.w3.org/feed/docs/rss2.html#ltpubdategtSubelementOfLtitemgt, the channel's last build date node is defined as

    "lastBuildDate 	The last time the content of the channel changed."

It should be possible to set the last build date within a custom built component.

**Comment by brianteeman** [comment]
this would require a big code change. You would need to load all the items, determine which is the newest and then use that date for the lastBuildDate - at this moment in time you are quite correct it always uses now https://github.com/joomla/joomla-cms/blob/staging/libraries/joomla/document/renderer/feed/rss.php#L81<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/13008">issues.joomla.org/tracker/joomla-cms/13008</a>.</sub>

**Comment by Ddcdom** [comment]
I understand that it would be a huge job to get the code working as per RSS specification. However, as pubDate is an optional channel element, wouldn't it be better to not include this in the feed output, rather than including incorrect information?

Allowing the pub date value to be changed in a component would be a smaller change and a big improvement over the current hard coded situation. This would allow custom components to manage the pubdate of their feed if they choose to.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/13008">issues.joomla.org/tracker/joomla-cms/13008</a>.</sub>

**Comment by franz-wohlkoenig** [comment]
> wouldn't it be better to not include this in the feed output, rather than including incorrect information?

I agree, @Ddcdom

**Comment by mbabker** [comment]
Fixed for 4.0.

In your `view.feed.php` file (or any place really before the document is rendered) you'll be able to do this:

```php
JFactory::getDocument()->lastBuildDate = new JDate('2017-02-04 17:11:00', new DateTimeZone('America/Chicago'));
```
